### PR TITLE
Test use @pingcap-incubator/pd-client-js lib

### PR DESCRIPTION
**Pull request by baurine**
- install and use @pingcap-incubator/pd-client-js from GitHub Packages
- update build scripts and related docs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -20,10 +20,10 @@ swagger:
 	scripts/generate_swagger.sh
 
 swagger_client: swagger
-	cd ui && yarn && npm run build_api_client
+	cd ui && yarn install --update-checksums && npm run build_api_client
```
Comment (breeswish) [inline Makefile]: Do we need `--frozen-lockfile` instead?

File: ui/src/index.js [hunk 2]
```diff
@@ -3,4 +3,12 @@ import ReactDOM from 'react-dom';
 import './index.css';
 import App from './App';
 
+import PDClient from '@pingcap-incubator/pd-client-js';
+
+// TODO
+const pdClient = new PDClient({
```
Comment (breeswish) [inline ui/src/index.js]: You can put it in `src/utils/client/index.js`, change it to expose two APIs:

```js
export default {
  dashboard: dashboardClient,
  pd: pdClient,
};
```

File: Makefile (hunk 1 above)
Comment (baurine) [reply Makefile]: Nope, adding `--update-checksums` is used to fix ["Integrity checked failed error"](https://github.com/yarnpkg/yarn/issues/7552) for yarn

File: ui/src/index.js (hunk 2 above)
Comment (baurine) [reply ui/src/index.js]: Ok, will fix it soon.

File: Makefile (hunk 1 above)
Comment (baurine) [reply Makefile]: It seems `--frozen-lockfile` option works fine as well.

**Comment by breeswish** [comment]
Nice. Looks like you need to update TiDB Dashboard's GitHub action file in order to make it work.

**Comment by baurine** [comment]
Yep, will fix it tmw, it seems we need to add a more step to create a `.npmrc` file for authentication.

**Comment by baurine** [comment]
Fix the github actions, now it works fine.

**Pull request merged**